cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java

```diff
@@ -271,8 +272,8 @@ private static void writeSynchronousMachines(Network network, String cimNamespac
             writer.writeCharacters(CgmesExportUtil.format(-g.getTargetQ()));
             writer.writeEndElement();
             writer.writeStartElement(cimNamespace, "SynchronousMachine.referencePriority");
-            // reference priority is used for angle reference selection (slack)
-            writer.writeCharacters(isInSlackBus(g) ? "1" : "0");
+            // reference priority is used for angle reference selection
+            writer.writeCharacters(Integer.toString(ReferencePriority.get(g)));
```

Comment:
although I don't think CGMES has batteries modelling... same change to do also for batteries further below, I'd expect the `isInSlackBus` method not to be used anywhere also

Reply:
done